client/my-sites/hosting/staging-site-card/index.js

```diff
@@ -116,27 +160,34 @@ export const StagingSiteCard = ( {
 				)
 			);
 		},
-		onSuccess: useCallback( () => {
-			dispatch(
-				successNotice( __( 'Staging site deleted.' ), { id: stagingSiteDeleteSuccessNoticeId } )
-			);
-		}, [ dispatch, __ ] ),
+		onSuccess: () => {
+			setProgress( 0.1 );
+		},
 	} );
-	const isStagingSiteTransferComplete = transferStatus === transferStates.COMPLETE;
+
 	useEffect( () => {
-		if ( wasCreating && isStagingSiteTransferComplete ) {
-			queryClient.invalidateQueries( {
-				queryKey: [ USE_SITE_EXCERPTS_QUERY_KEY ],
-			} );
+		if ( stagingSiteStatus === StagingSiteStatus.COMPLETE ) {
+			queryClient.invalidateQueries( [ USE_SITE_EXCERPTS_QUERY_KEY ] );
+			dispatch( setStagingSiteStatus( siteId, StagingSiteStatus.NONE ) );
 			dispatch(
 				successNotice( __( 'Staging site added.' ), { id: stagingSiteAddSuccessNoticeId } )
 			);
 		}
-	}, [ dispatch, queryClient, __, isStagingSiteTransferComplete, wasCreating ] );
+	}, [ dispatch, queryClient, __, isStagingSiteTransferComplete, siteId, stagingSiteStatus ] );
```

Comment:
I think we can remove `isStagingSiteTransferComplete` form the dependencies array:

```suggestion
	}, [ dispatch, queryClient, __, siteId, stagingSiteStatus ] );
```